osu.Game/Rulesets/Objects/HitObjectLifetimeEntry.cs

```diff
@@ -35,7 +35,11 @@ public HitObjectLifetimeEntry(HitObject hitObject)
             HitObject = hitObject;
 
             startTimeBindable.BindTo(HitObject.StartTimeBindable);
-            startTimeBindable.BindValueChanged(onStartTimeChanged, true);
+            startTimeBindable.BindValueChanged(_ => setInitialLifetime(), true);
+
+            // It is important to subscribe to this event before applied to a DrawableHitObject.
+            // Otherwise DHO cannot overwrite LifetimeStart set in setInitialLifetime.
```

Comment:
```suggestion
            // It is important to subscribe to this event before the DrawableHitObject so that the local callback is invoked before the entry is re-applied as a result of DefaultsApplied.
            // This way, the DrawableHitObject can use OnApply() to overwrite the LifetimeStart that was set inside setInitialLifetime().
```
I think this comment is trying to explain the ordering of events? This is my interpretation of it.

Reply:
Yes, your interpretation is correct.